drivers/adc/adc_ite_it8xxx2.c

```diff
@@ -180,40 +243,61 @@ static void adc_context_start_sampling(struct adc_context *ctx)
 {
 	struct adc_it8xxx2_data *data =
 		CONTAINER_OF(ctx, struct adc_it8xxx2_data, ctx);
+	uint32_t channels = ctx->sequence.channels;
+	uint8_t channel_count = 0;
+	int err;
 
 	data->repeat_buffer = data->buffer;
 
-	adc_enable_measurement(data->ch);
+	/*
+	 * The ADC sampling of it8xxx2 needs to read each channel
+	 * in sequence.
+	 */
+	while (channels) {
+		data->ch = find_lsb_set(channels) - 1;
+		channels &= ~BIT(data->ch);
+
+		adc_enable_measurement(data->ch);
+
+		channel_count++;
+	}
+
+	err = check_buffer_size(&ctx->sequence, channel_count);
+	if (err) {
```

Comment:
err is only used once, remove variable and put `check_buffer_size` into `if`

Reply:
Done.